src/main/java/jcifs/smb1/ntlmssp/Type1Message.java

```diff
@@ -0,0 +1,229 @@
+     * @throws IOException If an error occurs while parsing the material.
+     */
+    public Type1Message(byte[] material) throws IOException {
+        parse(material);
+    }
+
+    /**
+     * Returns the supplied authentication domain.
+     *
+     * @return A <code>String</code> containing the supplied domain.
+     */
+    public String getSuppliedDomain() {
+        return suppliedDomain;
+    }
+
+    /**
+     * Sets the supplied authentication domain for this message.
+     *
+     * @param suppliedDomain The supplied domain for this message.
+     */
+    public void setSuppliedDomain(String suppliedDomain) {
+        this.suppliedDomain = suppliedDomain;
+    }
+
+    /**
+     * Returns the supplied workstation name.
+     * 
+     * @return A <code>String</code> containing the supplied workstation name.
+     */
+    public String getSuppliedWorkstation() {
+        return suppliedWorkstation;
+    }
+
+    /**
+     * Sets the supplied workstation name for this message.
+     * 
+     * @param suppliedWorkstation The supplied workstation for this message.
+     */
+    public void setSuppliedWorkstation(String suppliedWorkstation) {
+        this.suppliedWorkstation = suppliedWorkstation;
+    }
+
+    public byte[] toByteArray() {
+        try {
+            String suppliedDomain = getSuppliedDomain();
+            String suppliedWorkstation = getSuppliedWorkstation();
+            int flags = getFlags();
+            boolean hostInfo = false;
+            byte[] domain = new byte[0];
+            if (suppliedDomain != null && suppliedDomain.length() != 0) {
+                hostInfo = true;
+                flags |= NTLMSSP_NEGOTIATE_OEM_DOMAIN_SUPPLIED;
+                domain = suppliedDomain.toUpperCase().getBytes(
+                        getOEMEncoding());
+            } else {
+                flags &= (NTLMSSP_NEGOTIATE_OEM_DOMAIN_SUPPLIED ^ 0xffffffff);
+            }
+            byte[] workstation = new byte[0];
+            if (suppliedWorkstation != null &&
+                    suppliedWorkstation.length() != 0) {
+                hostInfo = true;
+                flags |= NTLMSSP_NEGOTIATE_OEM_WORKSTATION_SUPPLIED;
+                workstation =
+                        suppliedWorkstation.toUpperCase().getBytes(
+                                getOEMEncoding());
+            } else {
+                flags &= (NTLMSSP_NEGOTIATE_OEM_WORKSTATION_SUPPLIED ^
+                        0xffffffff);
+            }
+            byte[] type1 = new byte[hostInfo ?
+                    (32 + domain.length + workstation.length) : 16];
+            System.arraycopy(NTLMSSP_SIGNATURE, 0, type1, 0, 8);
+            writeULong(type1, 8, 1);
+            writeULong(type1, 12, flags);
+            if (hostInfo) {
+                writeSecurityBuffer(type1, 16, 32, domain);
+                writeSecurityBuffer(type1, 24, 32 + domain.length, workstation);
+            }
+            return type1;
+        } catch (IOException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)